### ISSUE TITLE
Support any template engine

### ISSUE THREAD

**Issue by Toilal** [antfu/vite-plugin-md#10]
I have created a fork of this plugin to support any template engine instead of only markdown : https://github.com/Toilal/vite-plugin-md/tree/vite-plugin-render

The configuration is as below, you can use any template engine (markdown, pug, ...) through a custom render function.

```typescript
// vite.config.js
import Render from 'vite-plugin-md'
const MarkdownIt = require('markdown-it')

const markdown = new MarkdownIt({
  html: true,
  linkify: true,
  typographer: true
})

export default {
  plugins: [
    Vue({
      include: [/\.vue$/, /\.md$/], // <--
    }),
    Render({
      include: [/\.md$/],
      render: (content: string) => {
          return markdown.render(content)
      }
    }),
  ],
}
```

If you agree to make this plugin more generic, i'll open a pull request and let your rename the module to something more generic (vite-plugin-render maybe ?). Else I'll publish this fork on my own, please let met know.

**Comment by antfu** [comment]
Hi @Toilal, your plugin looks good! But TBH, I don't think we would make `vite-plugin-md` that general. Basically, this plugin offers much more than just transforming formats, keep it specific can also make it possible for us to optimize or add more features to it without requiring users to update their usages. 

Sorry as I would reject your request atm. But please keep up with your good work or even make more. Thank you.

**Comment by Toilal** [comment]
No problem at all, I was expecting this answer. i'll release my plugin and credit your work in the README.md, thanks.

**Comment by Toilal** [comment]
The plugin is released as 0.6.0 here : https://www.npmjs.com/package/vite-plugin-render

 i'll try to maintain it and merge your changes back :)
